rfqmsg/request.go

```diff
@@ -0,0 +1,472 @@
+	requestInAssetID = tlv.OptionalRecordT[tlv.TlvType5, asset.ID]
+
+	// requestInAssetGroupKey is a type alias for a record that represents
+	// the public group key of the inbound asset.
+	requestInAssetGroupKey = tlv.OptionalRecordT[
+		tlv.TlvType6, *btcec.PublicKey,
+	]
+
+	// requestOutAssetID is a type alias for a record that represents the
+	// asset ID of the outbound asset.
+	requestOutAssetID = tlv.OptionalRecordT[tlv.TlvType7, asset.ID]
+
+	// requestOutAssetGroupKey is a type alias for a record that represents
+	// the public group key of the outbound asset.
+	requestOutAssetGroupKey = tlv.OptionalRecordT[
+		tlv.TlvType8, *btcec.PublicKey,
+	]
+)
+
+// requestWireMsgData is a struct that represents the message data field for
+// a quote request wire message.
+type requestWireMsgData struct {
+	// Version is the version of the message data.
+	Version tlv.RecordT[tlv.TlvType0, WireMsgDataVersion]
+
+	// ID is the unique identifier of the quote request.
+	ID tlv.RecordT[tlv.TlvType1, ID]
+
+	// Expiry is the expiry Unix timestamp (in seconds) of the quote
+	// request. This timestamp defines the lifetime of both the suggested
+	// rate tick and the quote request.
+	Expiry tlv.RecordT[tlv.TlvType2, uint64]
+
+	// AssetMaxAmount represents the maximum asset amount that the target
+	// peer is expected to accept/divest.
+	AssetMaxAmount tlv.RecordT[tlv.TlvType3, uint64]
+
+	// SuggestedRateTick is the peer's proposed rate tick. This is not the
+	// final rate tick, but a suggested rate tick that the requesting peer
+	// would be willing to accept.
+	//
+	// NOTE: This field is optional.
+	SuggestedRateTick requestSuggestedTickRate
+
+	// InAssetID represents the identifier of the asset which will be
+	// inbound to the requesting peer (therefore outbound to the
+	// counterparty peer).
+	//
+	// NOTE: An asset ID of all zeros indicates BTC.
+	InAssetID requestInAssetID
+
+	// InAssetGroupKey is the public group key of the asset which will be
+	// inbound to the requesting peer (therefore outbound to the
+	// counterparty peer).
+	InAssetGroupKey requestInAssetGroupKey
+
+	// OutAssetID represents the identifier of the asset which will be
+	// outbound to the requesting peer (therefore inbound to the
+	// counterparty peer).
+	//
+	// NOTE: An asset ID of all zeros indicates BTC.
+	OutAssetID requestOutAssetID
+
+	// OutAssetGroupKey is the public group key of the asset which will be
+	// outbound to the requesting peer (therefore inbound to the
+	// counterparty peer).
+	OutAssetGroupKey requestOutAssetGroupKey
+}
+
+// newRequestWireMsgDataFromBuy creates a new requestWireMsgData from a buy
+// request.
+func newRequestWireMsgDataFromBuy(q BuyRequest) requestWireMsgData {
+	version := tlv.NewRecordT[tlv.TlvType0](q.Version)
+	id := tlv.NewRecordT[tlv.TlvType1](q.ID)
+
+	// Calculate the expiration unix timestamp in seconds.
+	expiry := tlv.NewPrimitiveRecord[tlv.TlvType2](
+		uint64(time.Now().Add(defaultRequestExpiry).Unix()),
```

Comment:
Kind of feels like this is the wrong place for this logic. Maybe we can set the expiry on the request already? But non-blocking as we don't really use the expiry yet anyway.

Reply:
I'll add a TODO to explain that this field should be populated on the `BuyRequest` and not just set here.